runhouse/resources/hardware/cluster_factory.py

```diff
@@ -95,260 +105,152 @@ def cluster(
         raise ValueError(
             "Cluster factory method can only accept one of `host` or `ips` as an argument."
         )
+    ips = kwargs.get("ips") or ([host] if isinstance(host, str) else host)
 
-    if name:
-        alt_options = dict(
-            host=host,
-            ssh_creds=ssh_creds,
-            server_port=server_port,
-            server_host=server_host,
-            server_connection_type=server_connection_type,
-            ssl_keyfile=ssl_keyfile,
-            ssl_certfile=ssl_certfile,
-            domain=domain,
-            den_auth=den_auth,
+    cluster_args = locals().copy()
+    cluster_args["host"] = kwargs.get("ips") or (
+        [host] if isinstance(host, str) else host
+    )
+    cluster_args["creds"] = cluster_args.pop("ssh_creds")
+    cluster_args.pop("kwargs")
+    cluster_args = {k: v for k, v in cluster_args.items() if v is not None}
+
+    # check for invalid args
+    valid_kwargs = ONDEMAND_COMPUTE_ARGS | KUBERNETES_CLUSTER_ARGS | RH_SERVER_ARGS
```

Comment:
this will break for python < 3.9 no? technically we still support 3.8, tho not sure we need to be

Reply:
oh good point, will edit